crates/rerun/src/web_viewer.rs

```diff
@@ -18,40 +20,58 @@ impl Drop for RemoteViewerServer {
 impl RemoteViewerServer {
     pub fn new(tokio_rt: &tokio::runtime::Runtime, open_browser: bool) -> Self {
         let (rerun_tx, rerun_rx) = re_smart_channel::smart_channel(re_smart_channel::Source::Sdk);
+        let (shutdown_tx, shutdown_rx_ws_server) = tokio::sync::broadcast::channel(1);
+        let shutdown_rx_web_server = shutdown_tx.subscribe();
 
         let web_server_join_handle = tokio_rt.spawn(async move {
             // This is the server which the web viewer will talk to:
             let ws_server = re_ws_comms::Server::new(re_ws_comms::DEFAULT_WS_SERVER_PORT)
                 .await
                 .unwrap();
-            let ws_server_handle = tokio::spawn(ws_server.listen(rerun_rx)); // TODO(emilk): use tokio_rt ?
+            let ws_server_handle = tokio::spawn(ws_server.listen(rerun_rx, shutdown_rx_ws_server)); // TODO(emilk): use tokio_rt ?
```

Comment:
So we really need this channel for the ws server? Doesn't calling `self.web_server_join_handle.abort()` (as we do in `drop`) already work?

Reply:
Hm. I could try ripping out the channel again from both servers and go with `abort`. Wonder if that is the same really as the shutdown I'm doing now.
I'd suggest not going there and unify as suggested above by using `shutdown_tx`on `Drop`

Reply:
dropping a handle does not cancel a task though, one needs to call `abort` on it explicitly? (only found this old thread where that decision was made, maybe has changed since then https://github.com/tokio-rs/tokio/issues/1830#issuecomment-696881238
Also do all tasks implement proper abort? The channel setup here follows tokio's recommendation on how to deal with ctrl+c

Reply:
I thought `abort` would work on any `async` task, since it simply stops the future from being polled.

Reply:
https://tokio.rs/tokio/topics/shutdown

```
let next_frame = tokio::select! {
    res = self.connection.read_frame() => res?,
    _ = self.shutdown.recv() => {
        // If a shutdown signal is received, return from `run`.
        // This will result in the task terminating.
        return Ok(());
    }
};
```

Maybe this is a bit too much out of context we're dealing with more specific tasks after all. Furthermore, most of this does boil down as an `abort`.
That said, I'm actually quite happy with this solution here since it gives us a single super easy entry point for signaling shut down which also makes it easy to work with hyper's `with_graceful_shutdown`